src/features/auth/Register/index.tsx

```diff
@@ -82,13 +74,15 @@ const Register = () => {
   }, []);
 
   const onFromSubmit = () => {
-    register(form).then((response: any) => {
-      const state = { isRegistered: !response.error };
-      const path = !response.error && ROUTES.AUTH.ROOT;
-      path ? navigate(path, { state }) : setIsEmailError(true);
-    });
+    register(form);
   };
 
+  useEffect(() => {
+    if (registerData && isSuccess) {
```

Comment:
I guess registerData is a bit redundant here. Probably "isSuccess" should be enough